glib/src/object.rs

```diff
@@ -288,6 +288,55 @@ pub trait Cast: ObjectType {
 
 impl<T: ObjectType> Cast for T {}
 
+// rustdoc-stripper-ignore-next
+/// Convenience trait mirroring `Cast`, implemented on `Option<Object>` types.
+pub trait CastNone: Sized {
+    type Inner;
+    fn downcast<T: ObjectType>(self) -> Result<T, Self>
```

Comment:
This seems like strange API. How would you distinguish `None` from failing to downcast now?

Similar question for `downcast_ref()` below. It should probably all returned `Option<T>` with another wrapping around?

Reply:
A failing downcast on `Some(t)` will return `Err(Some(t))`.

Reply:
Indeed on `downcast_ref()` currently you lose the error information, you can't distinguish why `None` was returned.
But hey, if the user wants to handle every conversion step, he can skip using this trait and keep doing whatever he is doing now.